### PR TITLE
[Material] Material theme builder

### PR DESCRIPTION
**Pull request by willlarche**
App showcasing how theming works in Material library. Created for the Flutter Interact event 2019.

### CODE REVIEW

File: material_theme_builder/lib/main.dart [hunk 1]
```diff
@@ -0,0 +1,537 @@
...
+      onPrimary: Color(0xFFFFFFFF),
+      onSecondary: Color(0xFF000000),
+      onSurface: Color(0xFF000000),
+      onError: Color(0xFFFFFFFF),
+      onBackground: Color(0xFF000000),
+      brightness: Brightness.light,
+    );
+
+    final TextTheme textTheme = TextTheme(
+      display4: GoogleFonts.roboto(
```
Comment (guidezpl) [inline material_theme_builder/lib/main.dart]: suggestion: remove trailing fontSize comma

File: material_theme_builder/lib/main.dart [hunk 2]
```diff
@@ -0,0 +1,537 @@
...
+                      ),
+                    ),
+                  ]),
+            ),
+            SizedBox(height: 32),
+
+            // TEXT FIELD
+            Header('TEXT FIELD'),
+
+            // This was in the original design but applying theming doesn't work
```
Comment (guidezpl) [inline material_theme_builder/lib/main.dart]: remove commented out code

File: material_theme_builder/pubspec.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
+name: material_theme_builder
+description: A new Flutter project.
```
Comment (guidezpl) [inline material_theme_builder/pubspec.yaml]: update description

File: material_theme_builder/lib/main.dart [hunk 4]
```diff
@@ -0,0 +1,537 @@
+import 'package:flutter/material.dart';
```
Comment (perclasson) [inline material_theme_builder/lib/main.dart]: Add license here.

File: material_theme_builder/lib/adaptive.dart [hunk 5]
```diff
@@ -0,0 +1,28 @@
+// Copyright 2019 The Flutter team. All rights reserved.
```
Comment (perclasson) [inline material_theme_builder/lib/adaptive.dart]: This file is not used, do we still intend to have it here?

File: material_theme_builder/README.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
+# Build a Material Theme
```
Comment (RedBrogdon) [inline material_theme_builder/README.md]: We have a general format for these READMEs that we like to use (goals, important bits, etc.). You can see it in action here:

https://github.com/flutter/samples/tree/master/isolate_example

File: material_theme_builder/android/app/src/main/AndroidManifest.xml [hunk 7]
```diff
@@ -0,0 +1,30 @@
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="com.example.material_theme_builder">
```
Comment (RedBrogdon) [inline material_theme_builder/android/app/src/main/AndroidManifest.xml]: This is a nit, but I'm trying to keep the package names for samples relatively cohesive. This should be "dev.flutter.examples", and you can do that update with this command from the project root:

    flutter create --org=dev.flutter.examples .

**Comment by RedBrogdon** [comment]
@willlarche is this still in progress?

**Comment by domesticmouse** [comment]
Hey @willlarche can we land this please?

File: .metadata [hunk 8]
```diff
@@ -0,0 +1,10 @@
+# This file tracks properties of this Flutter project.
```
[the remark this answers is not in the document]
Comment (willlarche) [reply .metadata:1]: They keep them in this repo.

File: INDEX.md [hunk 9]
```diff
@@ -106,6 +106,12 @@ A collection of animation examples. The app starts with basic concepts like
 `AnimationController` and `Tween`, then progresses to transitions and
 advanced demos.
 
+#### [Material Theme Builder]l(material_theme_builder) _(Material team)_
+
+Build a Material Theme lets you create your own Material theme by customizing values for color, 
+typography, and shape. See how these values appear when applied to Material Components and
```
[the remark this answers is not in the document]
Comment (willlarche) [reply INDEX.md:112]: Done.

File: material_theme_builder/README.md [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+
+Build a Material Theme lets you create your own Material theme by customizing values for color, typography, and shape. See how these values appear when applied to Material Components and discover how to implement your custom theme in your own projects.
```
[the remark this answers is not in the document]
Comment (willlarche) [reply material_theme_builder/README.md:3]: Done.

File: material_theme_builder/README.md [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+
+## Getting Started
+
+This project is a starting point for building a Material Theme on Flutter. Learn how to customize Material in this talk [Material Theming with Flutter](https://youtu.be/stoJpMeS5aY).
```
[the remark this answers is not in the document]
Comment (willlarche) [reply material_theme_builder/README.md:7]: Done.

File: material_theme_builder/lib/main.dart (hunk 1 above)
Comment (willlarche) [reply material_theme_builder/lib/main.dart:27]: Done.

File: material_theme_builder/pubspec.yaml (hunk 3 above)
Comment (willlarche) [reply material_theme_builder/pubspec.yaml:2]: Done.

File: material_theme_builder/lib/adaptive.dart (hunk 5 above)
Comment (willlarche) [reply material_theme_builder/lib/adaptive.dart:1]: Deleted.

File: material_theme_builder/README.md (hunk 6 above)
Comment (willlarche) [reply material_theme_builder/README.md:1]: Done.

Comment (willlarche) [reply material_theme_builder/README.md:1]: Please re review that file.

File: INDEX.md [hunk 12]
```diff
@@ -106,6 +106,12 @@ A collection of animation examples. The app starts with basic concepts like
 `AnimationController` and `Tween`, then progresses to transitions and
 advanced demos.
 
+#### [Material Theme Builder]l(material_theme_builder) _(Material team)_
+
+Build a Material Theme lets you create your own Material theme by customizing values for color,
```
[the remark this answers is not in the document]
Comment (willlarche) [reply INDEX.md]: Done.

File: material_theme_builder/lib/main.dart (hunk 2 above)
Comment (willlarche) [reply material_theme_builder/lib/main.dart:434]: Done.

File: material_theme_builder/android/app/src/main/AndroidManifest.xml (hunk 7 above)
Comment (willlarche) [reply material_theme_builder/android/app/src/main/AndroidManifest.xml:2]: Done.

File: material_theme_builder/lib/main.dart (hunk 4 above)
Comment (willlarche) [reply material_theme_builder/lib/main.dart:1]: Done.

**Pull request merged**